### ISSUE TITLE
Can't reproduce OE results

### ISSUE THREAD

**Issue by ivan-chai** [Jingkang50/OpenOOD#182]
Hello!

I try to reproduce OE results on ImageNet200 using ```scripts/ood/oe/imagnet200_train_oe.sh```, but get 55% accuracy on the validation set. Test set accuracy, as reported in the paper, must be about 86%. Does the repo contains actual training scripts for OE? What validation accuracy do you get in this setup?

**Comment by zjysteven** [comment]
Those scripts are exactly the ones that we used to run the experiments and to obtain the results in the paper. Would you try using just 1 GPU instead to see if it's because of the DDP sampler issue you mentioned? We were using 2 GPUs just to speed up training. Also, although it's very unlikely, please make sure your code is up-to-date with the repo; in earlier version there are indeed some bugs with the OE implementation that caused severe performance degradation, as mentioned in #138.

**Comment by ivan-chai** [comment]
Thank you for recommendation. I'll try to check multi-GPU training. Is ImageNet 200 dataset is just a subset of ImageNet 1-k? Or there are preprocessing differences?

**Comment by zjysteven** [comment]
Yes it is just a subset of ImageNet1k. It uses the common preprocessing of ImageNet (ResizedCrop, Flipping, Normalization) and has nothing fancy/special.

**Comment by ivan-chai** [comment]
I evaluated the model trained with 2 GPUs. Results are attached. Test set accuracy is much larger than validation. However, metrics are low compared to the paper.
<img width="623" alt="Screenshot 2023-09-12 at 15 07 36" src="https://github.com/Jingkang50/OpenOOD/assets/15258512/ce85a1dc-fd3a-417d-9aa7-5197f1e04f81">

**Comment by zjysteven** [comment]
Hi @ivan-chai, I retrained two OE models with the following commands and here are the results I got.

-----
Training with 2 gpus:
```bash
CUDA_VISIBLE_DEVICES="1,2" \
python main.py \
    --config configs/datasets/imagenet200/imagenet200.yml \
    configs/datasets/imagenet200/imagenet200_oe.yml \
    configs/networks/resnet18_224x224.yml \
    configs/pipelines/train/baseline.yml \
    configs/pipelines/train/train_oe.yml \
    configs/preprocessors/base_preprocessor.yml \
    --optimizer.num_epochs 90 \
    --dataset.train.batch_size 128 \
    --num_gpus 2 --num_workers 16 \
    --merge_option merge \
    --seed 4
```
![image](https://github.com/Jingkang50/OpenOOD/assets/19763611/f5d8a2c4-1e77-43b3-91fa-8e4dcaa6b56a)

-----

Training with 1 gpu:
```bash
CUDA_VISIBLE_DEVICES=3 \
python main.py \
    --config configs/datasets/imagenet200/imagenet200.yml \
    configs/datasets/imagenet200/imagenet200_oe.yml \
    configs/networks/resnet18_224x224.yml \
    configs/pipelines/train/baseline.yml \
    configs/pipelines/train/train_oe.yml \
    configs/preprocessors/base_preprocessor.yml \
    --optimizer.num_epochs 90 \
    --dataset.train.batch_size 256 \
    --num_gpus 1 --num_workers 16 \
    --merge_option merge \
    --seed 5
```
![image](https://github.com/Jingkang50/OpenOOD/assets/19763611/04c8e29d-dc09-4d9f-8595-47fef9231c91)

------

These results are quite close to what we reported in the paper.

**Comment by ivan-chai** [comment]
Thank you for the provided results!

I'll try to run on a different machine and environment.
